### PR TITLE
[Proposal] Introduce vue-class-component instead of av-ts

### PR DESCRIPTION
**Pull request by riseshia**
Changed:

- Remove babel
- Introduce vue-property-decorator
- Cleaning some code

### CODE REVIEW

File: build/webpack.base.conf.js [hunk 1]
```diff
@@ -44,9 +44,14 @@ module.exports = {
         include: [resolve('src'), resolve('test')]
       },
       {
-        test: /\.tsx?$/,
+        test: /\.ts$/,
```
Comment (riseshia) [inline build/webpack.base.conf.js:47]: we don't have plan to use tsx, so it's better to remove I think

File: package.json [hunk 2]
```diff
@@ -73,6 +64,7 @@
     "typescript": "^2.1.6",
     "url-loader": "^0.5.7",
     "vue-loader": "^11.0.0",
+    "vue-property-decorator": "^3.4.0",
```
Comment (riseshia) [inline package.json]: it requires `vue-class-component`

File: src/components/Todo.vue [hunk 3]
```diff
@@ -10,22 +10,28 @@
...
-  @Prop todoId = p({ type: Number, required: true })
-  @Prop description = p({ type: String, required: true })
-  @Prop type = p({ type: String, required: true })
-  @Prop estimatedSize = p({ type: Number, required: true })
-  @Prop status = p({ type: Boolean, required: true })
-
   instanceStatus = this.status
 
+  @Prop({ required: true })
+  todoId: number
```
Comment (riseshia) [inline src/components/Todo.vue]: To me, it's better than `av-ts` with below reason.

- it's more natural with typescript style
- there is no magic method like `p`, all magic happened in decorator.

File: src/main.ts [hunk 4]
```diff
@@ -17,7 +17,6 @@ declare module "vue/types/options" {
   }
 }
 
-/* eslint-disable no-new */
```
Comment (riseshia) [inline src/main.ts:20]: we don't use eslint XD

File: test/e2e/nightwatch.conf.js [hunk 5]
```diff
@@ -1,4 +1,3 @@
-require('babel-register')
```
Comment (riseshia) [inline test/e2e/nightwatch.conf.js:1]: it seems not to affect

File: test/unit/specs/Todo.spec.ts [hunk 6]
```diff
@@ -10,7 +10,7 @@ describe("Todo.vue", () => {
       }
     }).$mount();
 
-    const actual = vm.$el.textContent;
+    const actual = <string> vm.$el.textContent;
```
Comment (riseshia) [inline test/unit/specs/Todo.spec.ts:13]: textContent is `string | null` so we need to cast to `string` for passing to `assert.include`

File: tsconfig.json [hunk 7]
```diff
@@ -1,13 +1,18 @@
 {
   "compilerOptions": {
-    "module": "commonjs",
     "target": "es5",
-    "sourceMap": true,
-    "allowJs": true,
+    "lib": [
+      "dom",
+      "es2015"
+    ],
```
Comment (marocchino) [inline tsconfig.json]: see line 13

Comment (riseshia) [reply tsconfig.json]: ( ﾟдﾟ)!!!!! I will remove

**Review by marocchino — APPROVED** [review]

**Pull request merged**